src/logger.ts

```diff
@@ -40,42 +32,80 @@ function getTimestamp(): string {
 
 export class Logger implements Disposable {
     private disposables: Disposable[] = [];
-    private fd = 0;
+    private fd: number | undefined;
+    private filePath: string | undefined;
     private loggers: Map<string, ILogger> = new Map();
-    private level!: LogLevel;
+    private level!: vscode.LogLevel;
     private logToConsole!: boolean;
     private outputChannel?: vscode.LogOutputChannel;
 
     /**
      * Setup logging for the extension. Logs are dropped unless one or more of
      * `filePath`, `logToConsole`, or `outputChannel` is given.
      *
-     * @param level Only log messages at or above this level, or never if set to `LogLevel.none`.
+     * @param level Only log messages at or above this level, or never if set to `vscode.LogLevel.Off`.
      * @param filePath Write messages to this file.
      * @param logToConsole Write messages to the `console` (Hint: run the "Developer: Toggle Developer Tools" vscode command to see the console).
      * @param outputChannel Write messages to this vscode output channel.
      */
-    public init(level: LogLevel, filePath: string, logToConsole = false, outputChannel?: vscode.LogOutputChannel) {
+    public init(
+        level: vscode.LogLevel,
+        filePath: string,
+        logToConsole = false,
+        outputChannel?: vscode.LogOutputChannel,
+    ) {
         this.level = level;
         this.logToConsole = logToConsole;
         this.outputChannel = outputChannel;
-        if (filePath && level !== LogLevel.none) {
-            try {
-                this.fd = fs.openSync(filePath, "w");
-                this.disposables.push({
-                    dispose: () => fs.closeSync(this.fd),
-                });
-            } catch {
-                // ignore
-            }
-        }
+        this.filePath = filePath;
+
+        this.listenForLogLevelChanges();
+        this.setupLogFile();
     }
 
     public dispose(): void {
         disposeAll(this.disposables);
     }
 
-    private log(level: LogLevel, scope: string, logToOutputChannel: boolean, args: any[]): void {
+    private listenForLogLevelChanges() {
```

Comment:
don't we use the `on` prefix for listeners?
```suggestion
    private onLogLevelChanged() {
```

Reply:
This isn't a listener, but rather sets one up. That said I could move the subscription into the constructor and then rename this; that's definitely more consistent with how this project does things.